### PR TITLE
feat: add support for XEP-0333

### PR DESCRIPTION
**Pull request by marc0s**
This is my attempt to support Chat Markers in stanza.io.  Will do some more testing, but this seems to be working in our environment.

Feedback will be welcomed :)

### CODE REVIEW

File: lib/plugins/markers.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+module.exports = function (client, stanzas, config) {
+
+	function enabled(msg) {
+		return msg.markable && config.chatMarkers !== false;
```
Comment (legastero) [inline lib/plugins/markers.js]: Best to check client.config here instead so that it can be changed on the fly.

Comment (legastero) [reply lib/plugins/markers.js]:
```suggestion
		return msg.markable && client.config.chatMarkers !== false;
```

**Comment by parasmn81** [comment]
In order to have a message marked as "markable", e.g.
`<message>`
`<body>...</body>`
`<markable xmlns='urn:xmpp:chat-markers:0'/>`
`</message>`

I had to "enable"  it on client.sendMessage({..}); i.e

`const msg = {
                    to: 'recipient@domain',
                    body: 'message',
                    markable: true,
                } 
  const id = this.xmppClient.sendMessage(msg);`

Couldn't be enabled directly by checking the "chatMarkers" value passed in the options: `chatMarkers: true,` ?

**Comment by marc0s** [comment]
> Couldn't be enabled directly by checking the "chatMarkers" value passed in the options: chatMarkers: true, ?

Might be, but I'm not sure you'll want *all* your messages as markable.  You'll probably better with some wrapper around stanza for your use case like in `sendMarkableMessage = (to, body) => client.sendMessage({to, body, markable: true})`.

**Comment by parasmn81** [comment]
Indeed, sending markable messages to all clients is too much.
However, it was a bit frustrating to enable it and discover that it doesn't work. (could be added to the api reference).
The xep-0333 mentions inspecting the client caps.. in order to enable markable . I'll have a look to see what is the stanza support to "store" this for each bare jid.

P.S. thank you for the implementation and quick reply.

**Pull request merged**